### PR TITLE
Docstring and call signature modifications

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [ ] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* cleaner logic in `indices._anuclim._to_quarter()`
* many docstring style corrections
* more statically-typed function call signatures
* consistency in style

### Does this PR introduce a breaking change?

No.

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@aulemahal Thanks for the review! There's maybe one or two final adjustments to make from what I can see, but I've identified several `# FIXME` issues (mostly unused call signatures). Would it make sense to remove some of these in this PR, or do you think that might potentially break downstream services?

File: xclim/analog.py [hunk 1]
```diff
@@ -72,9 +72,10 @@
 # based on distances. J Stat Planning & Inference 143: 1249-1272
 
 # TODO: Hellinger distance
-metrics = {}
+metrics = dict()
 
 
+# FIXME: Kwargs seem not to be passed to any functions.
```
Comment (aulemahal) [inline xclim/analog.py:78]: Woups. True, this could be fixed here by passing it in the apply_ufunc call.

File: xclim/indices/_anuclim.py [hunk 2]
```diff
@@ -403,6 +403,7 @@ def prcptot_warmcold_quarter(
     return out
 
 
+# FIXME: `src_timestep` is not used here.
```
Comment (aulemahal) [inline xclim/indices/_anuclim.py:406]: These two FIXME could also be fixed here. But if removing the arg creates bugs in the tests, we can wait another PR.

File: xclim/indices/_multivariate.py [hunk 3]
```diff
@@ -117,6 +117,8 @@ def cold_spell_duration_index(
     >>> tn10 = percentile_doy(tasmin, per=10).sel(percentiles=10)
     >>> cold_spell_duration_index(tasmin, tn10)
     """
+    # FIXME: Bootsrap flag is not used. Should be removed.
```
Comment (aulemahal) [inline xclim/indices/_multivariate.py:120]: It is used by the decorator!

File: xclim/indices/_anuclim.py (hunk 2 above)
Comment (Zeitsperre) [reply xclim/indices/_anuclim.py:406]: Alright, let's keep the FIXME's there and wait for another PR.

**Pull request merged**